src/components/favorite-button/favorite-button.tsx

```diff
@@ -2,12 +2,19 @@ import { FC, useState } from "react";
 import { IFavoriteButtonProps } from "./favorite-button.interface";
 import { FavoriteWrapper } from "./favorite-button.styles";
 import { AiFillHeart } from "react-icons/ai";
+import { useSession } from "next-auth/react";
 
-export const FavoriteButton: FC<IFavoriteButtonProps> = () => {
+export const FavoriteButton: FC<IFavoriteButtonProps> = ({
+  openLoginModal,
+}) => {
   const [isLiked, setIsLiked] = useState<boolean>(false);
+  const { data } = useSession();
```

Comment:
🦆 this `useSession` should not be included in the UI. When the consumer of the favorite-button, sets the `onClick`, this will consume the data from the useSession. In a similar way, this component can receive a boolean prop called, `isLiked`, which can be used to control the visual state of the component from the outside

Reply:
🥔 I have removed the useSession from this component and move it to the parent and send it as props instead. Same with the function, I have moved it to the parent 🔥